presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIndexedQueries.java

```diff
@@ -640,4 +655,11 @@ public void testNonEquiJoin()
     {
         assertQuery("SELECT COUNT(*) FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND lineitem.quantity + length(orders.comment) > 7");
     }
+
+    private void assertExplainAnalyze(@Language("SQL") String query)
+    {
+        String value = (String) computeActual(query).getOnlyValue();
+
+        assertTrue(value.matches("(?s:.*)CPU:.*, Input:.*, Output(?s:.*)"), format("Expected output to contain \"CPU:.*, Input:.*, Output\", but it is %s", value));
```

Comment:
This is very brittle. The format of explain analyze is not guaranteed or even expected to be stable.

Reply:
Yeah, I just took from the AbstractTestQueries :) . It basically only make sure the query does not fail. Can we return explain result as json format? that way, it will be easier to test and also easier for user client to parse.

Reply:
Explain is not designed to be machine readable.

Reply:
> Can we return explain result as json format

That would have similar problems. The plan IR is not stable, so that could cause the json-formatted version to change from version to version.

Reply:
I would just validate that results are returned, rather than an error.